CoEpi/ui/onboarding/AttributedString.swift

```diff
@@ -0,0 +1,71 @@
+import UIKit
+
+typealias FontAttributes = [NSAttributedString.Key: Any]
+
+struct AttributableString {
+    
+    let baseFont: UIFont
+    var storage = NSMutableAttributedString()
+    
+    let boldAttributes: FontAttributes
+    let normalAttributes: FontAttributes
+
+    init(baseFont: UIFont) {
+        
+        self.baseFont = baseFont
+        
+        let pointSize = baseFont.pointSize
+        let fontDescriptor = baseFont.fontDescriptor.withSymbolicTraits(.traitBold)
+            ?? UIFont.boldSystemFont(ofSize: pointSize).fontDescriptor
+        
+        let boldFont = UIFont(descriptor: fontDescriptor, size: pointSize)
+        
+        self.boldAttributes = [
+            .font: boldFont
+        ]
+        
+        self.normalAttributes = [
+            .font: baseFont
+        ]
+        
+    }
+    
+    func append(_ value: String, attributes: FontAttributes) {
+        let attributedString = NSAttributedString(string: "\(value) ", attributes: attributes)
+        storage.append(attributedString)
+    }
+    
+    func bold(_ value: String) -> AttributableString {
+        
+        append(value, attributes: boldAttributes)
+        
+        return self
+        
+    }
+    
+    func normal(_ value: String) -> AttributableString {
+        
+        append(value, attributes: normalAttributes)
+        
+        return self
+        
+    }
+    
+    func finish() -> NSAttributedString {
+        
+        let range = NSRange(location: 0, length: storage.length - 1)
+        return storage.attributedSubstring(from: range)
```

Comment:
Is this a defensive copy (otherwise it seems fine to just return `storage`)?

Reply:
Returning the substring is to remove the trailing space that will be added as above. Couple of ways I see round this, build the string by pushing to the end of an array of `NSAttributedString` instead and doing join in the final method. But in the final commit this class will be likely removed. I will make the change and keep the code just in case it becomes useful later. I see the major problem with it being translating the emphasis (bold etc) into right to left languages. See the next commit for the altered solution

Reply:
Another reason to not add those spaces in `append` 😉